src/flows/components/panel/PanelVisibilityToggle.tsx

```diff
@@ -24,8 +24,17 @@ const PanelVisibilityToggle: FC<Props> = ({id}) => {
       state: !meta.visible ? 'true' : 'false',
     })
 
-    flow.meta.update(id, {
-      visible: !meta.visible,
+    update({
+      meta: {
+        ...flow.meta,
+        byID: {
+          ...flow.meta.byID,
+          [id]: {
+            ...flow.meta.byID[id],
+            visible: !meta.visible,
+          },
+        },
+      },
```

Comment:
I feel like top-level we can just do something like `update(prev => {...prev, allThatOtherStuff})` and that might reduce some of the lifecycle stuff?